src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -314,98 +431,184 @@ public enum EvalMetricType
+        /// Whether to enable categorical split or not.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Enable categorical split or not.", ShortName = "cat")]
         [TlcModule.SweepableDiscreteParam("UseCat", new object[] { true, false })]
         public bool? UseCategoricalSplit;
 
+        /// <summary>
+        /// Whether to enable special handling of missing value or not.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Enable special handling of missing value or not.")]
         [TlcModule.SweepableDiscreteParam("UseMissing", new object[] { true, false })]
         public bool HandleMissingValue = false;
 
+        /// <summary>
+        /// The minimum number of data points per categorical group.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Minimum number of instances per categorical group.", ShortName = "mdpg")]
         [TlcModule.Range(Inf = 0, Max = int.MaxValue)]
         [TlcModule.SweepableDiscreteParam("MinDataPerGroup", new object[] { 10, 50, 100, 200 })]
         public int MinimumExampleCountPerGroup = 100;
 
+        /// <summary>
+        /// When the number of categories of one feature is smaller than or equal to <see cref="MaximumCategoricalSplitPointCount"/>,
+        /// one-vs-other split algorithm will be used.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Max number of categorical thresholds.", ShortName = "maxcat")]
         [TlcModule.Range(Inf = 0, Max = int.MaxValue)]
         [TlcModule.SweepableDiscreteParam("MaxCatThreshold", new object[] { 8, 16, 32, 64 })]
         public int MaximumCategoricalSplitPointCount = 32;
 
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Lapalace smooth term in categorical feature spilt. Avoid the bias of small categories.")]
+        /// <summary>
+        /// Laplace smooth term in categorical feature split.
+        /// This can reduce the effect of noises in categorical features, especially for categories with few data.
+        /// </summary>
+        /// <value>
+        /// Constraints: <see cref="CategoricalSmoothing"/> >= 0.0
+        /// </value>
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Laplace smooth term in categorical feature split. Avoid the bias of small categories.")]
         [TlcModule.Range(Min = 0.0)]
         [TlcModule.SweepableDiscreteParam("CatSmooth", new object[] { 1, 10, 20 })]
         public double CategoricalSmoothing = 10;
 
+        /// <summary>
+        /// L2 regularization for categorical split.
+        /// </summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "L2 Regularization for categorical split.")]
         [TlcModule.Range(Min = 0.0)]
         [TlcModule.SweepableDiscreteParam("CatL2", new object[] { 0.1, 0.5, 1, 5, 10 })]
         public double L2CategoricalRegularization = 10;
 
+        /// <summary>
+        /// The random seed for LightGBM to use.
```

Comment:
>The random seed for LightGBM to use. [](start = 12, length = 36)

Please specify what happens if the seed is not set by the user.

Reply:
done

---
In reply to: [263946446](https://github.com/dotnet/machinelearning/pull/2886#discussion_r263946446) [](ancestors = 263946446)